context-propagation/src/main/java/io/micrometer/context/ContextExecutorService.java

```diff
@@ -150,13 +150,29 @@ public static ExecutorService wrap(ExecutorService service, Supplier<ContextSnap
         return new ContextExecutorService<>(service, snapshotSupplier);
     }
 
+    /**
+     * Wrap the given {@code ExecutorService} in order to propagate context to any
+     * executed task through the given {@link ContextSnapshot} supplier.
+     * @param service the executorService to wrap
+     * @param clearMissing whether to {@link ThreadLocalAccessor#setValue() clear}
+     * {@link ThreadLocal} values not captured upon task submission
+     * @return wrapped instance
+     * @since 1.0.3
+     */
+    public static ExecutorService wrap(ExecutorService service, boolean clearMissing) {
+        return new ContextExecutorService<>(service, () -> DefaultContextSnapshotFactory
+            .captureAll(ContextRegistry.getInstance(), key -> true, clearMissing));
+    }
+
```

Comment:
Maybe we can avoid a convenience method here, which can open the door for even more such methods, not unlike all the variations just deprecated in `ContextSnapshot`. Essentially leave keep such customizations at the level of `ContextSnapshotFactory`.